src/devices/stepper.rs

```diff
@@ -0,0 +1,526 @@
+    Run,
+}
+
+impl Stepper {
+    /// Create a new Stepper sensor.
+    pub fn new() -> Self {
+        let mut chan: StepperHandle = ptr::null_mut();
+        unsafe {
+            ffi::PhidgetStepper_create(&mut chan);
+        }
+        Self::from(chan)
+    }
+
+    /// Get a reference to the underlying sensor handle
+    pub fn as_channel(&self) -> &StepperHandle {
+        &self.chan
+    }
+
+    /// Set enable failsafe
+    pub fn set_enable_failsafe(&self, failsafe_time: u32) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_enableFailsafe(self.chan, failsafe_time)
+        })?;
+        Ok(())
+    }
+
+    /// Add position offset
+    pub fn get_add_position_offset(&self, position_offset: f64) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_addPositionOffset(self.chan, position_offset)
+        })?;
+        Ok(())
+    }
+
+    /// Set reset failsafe
+    pub fn set_reset_failsafe(&self) -> Result<()> {
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_resetFailsafe(self.chan) })?;
+        Ok(())
+    }
+    /// Set acceleration
+    pub fn set_acceleration(&self, acceleration: f64) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_setAcceleration(self.chan, acceleration)
+        })?;
+        Ok(())
+    }
+
+    /// Get acceleration
+    pub fn get_acceleration(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getAcceleration(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get minimum acceleration
+    pub fn get_min_acceleration(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMinAcceleration(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Get maximum acceleration
+    pub fn get_max_acceleration(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMaxAcceleration(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Set control mode
+    pub fn set_control_mode(&self, control_mode: ControlMode) -> Result<()> {
+        let _cm = match control_mode {
+            ControlMode::Step => 0,
+            ControlMode::Run => 1,
+        };
+
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_setControlMode(self.chan, _cm) })?;
+        Ok(())
+    }
+
+    /// Get control mode
+    pub fn get_control_mode(&self) -> Result<ControlMode> {
+        let mut _cm: ffi::PhidgetStepper_ControlMode = 0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getControlMode(self.chan, &mut _cm) })?;
+
+        match _cm {
+            1 => Ok(ControlMode::Run),
+            0 => Ok(ControlMode::Step),
+            _ => Err(ReturnCode::UnknownVal),
+        }
```

Comment:
Nit: Probably better to implement this as the `TryFrom` trait for the `ControlMode` type. Keep it all together.

Like:

```
impl TryFrom<u32> for ControlMode {
    type Error = crate::Error;

    fn try_from(val: u32) -> Result<Self, Self::Error> {
        using ControlMode::*;
        match val {
            0 => Ok(Step),
            1 => Ok(Run),
            _ => Err(ReturnCode::UnknownVal),
        }
    }
}
```

Reply:
good idea